File: 278. First Bad Version.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return a bool
# def isBadVersion(version):

class Solution(object):
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        start = 1
        end = n
        
        while start <= end:
            mid = (start + end)/2
            if not isBadVersion(mid):
                start = mid + 1
            else:
                end = mid - 1
        return start

File: 278. First Bad Version.cpp
// timeout

// Forward declaration of isBadVersion API.
bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int start = 1;
        int end = n;
        while (start <= end)
        {
            int mid = (start + end)/2;
            if (!isBadVersion(mid))
                start = mid + 1;
            else
                end = mid - 1;
        }
        return start;
    }
};
